src/rt/util/random.d

```diff
@@ -12,10 +12,47 @@ struct Rand48
 
 @safe @nogc nothrow:
 
-    void defaultSeed()
+    void defaultSeed() @trusted
     {
-        import ctime = core.stdc.time : time;
-        seed(cast(uint)ctime.time(null));
+        version(D_InlineAsm_X86_64)
+        {
+            ulong result = void; // Workaround for LDC issue #950, cannot access struct members in DMD asm.
+            asm @nogc nothrow
+            {
+                rdtsc;
+                // RAX: low 32 bits are low bits of timestamp, high 32 bits are 0.
+                // RDX: low 32 bits are high bits of timestamp, high 32 bits are 0.
+                // We combine these into a 48 bit value instead of a full 64 bits
+                // because `front` and `popFront` only make use of the bottom 48
+                // bits of `rng_state`.
+                shl RDX, 16;
+                xor RDX, RAX;
+                mov result, RDX; // result: low 48 bits are ???, high 16 bits are 0.
+            }
+            rng_state = result;
+            popFront();
+        }
+        //else version(D_InlineAsm_X86)
+        //{
+        //    // We don't use `rdtsc` with version(D_InlineAsm_X86) because
+        //    // some x86 processors don't support `rdtsc` and because on
+        //    // x86 (but not x86-64) Linux `prctl` can disable a process's
+        //    // ability to use `rdtsc`.
+        //    static assert(0);
+        //}
+        else version(OSX)
+        {
+            // mach_absolute_time is much faster than ctime.time.
+            import core.time : mach_absolute_time;
+            rng_state = mach_absolute_time();
+            popFront();
+        }
+        else
+        {
+            // Old version using timestamp in seconds:
```

Comment:
Old version?

Reply:
Current version.

Reply:
Perhaps you mean libc or fallback version then. Old version sounds like you intend to remove it.

Reply:
Changed comment as suggested.